damnit/ctxsupport/damnit_ctx.py

```diff
@@ -64,6 +65,12 @@ def check(self):
             problems.append(
                 f"data={self._data!r} for variable {self.name} (can be 'raw'/'proc')"
             )
+        if self.tags is not None:
+            if not isinstance(self.tags, Sequence) or not all(isinstance(tag, str) for tag in self.tags):
+                problems.append(
+                    f"tags={self.tags!r} for variable {self.name} (must be a string or an iterable of strings)"
```

Comment:
```suggestion
            if not isinstance(self.tags, Sequence) or not all(isinstance(tag, str) and tag != "" for tag in self.tags):
                problems.append(
                    f"tags={self.tags!r} for variable {self.name} (must be a non-empty string or an iterable of strings)"
```

I thought of this after doing the review. I think it would be a bit confusing having an empty tag around.

If you agree with this I would also add a test to check for this case.